src/objectid.ts

```diff
@@ -103,17 +133,38 @@ export class ObjectId extends BSONValue {
       workingId = inputId;
     }
 
+    // If we have reached the end of the pool then create a new pool
+    if (!pool || poolOffset + 12 > pool.byteLength) {
+      createPool();
+    }
+    this[_pool] = pool;
+    this[_offset] = poolOffset;
+    poolOffset += 12;
```

Comment:
Above I suggest changing the type of the module scoped pool to include `null` which forces us to check here if pool is `null`. I'm thinking we can migrate the condition to create a new pool into `createPool` and possibly call it `getPool` instead. The method can either return the existing a pool or a new one if adding 12 bytes goes beyond the byteLength. 

My motive is to add a bit more type truth to that global pool since we want to be sure to start the module with the pool being `null` so that users have a chance to adjust the poolSize before any ObjectIds implicitly create it.

Reply:
Great idea. I was thinking something similar with a `getPool` which would handle creating the pool if necessary too. There is a *small* issue with the code above currently where calling the constructor with an invalid value (ie `new ObjectId('abc')`) would still increment the poolOffset, leading to wasted space in the pool. Thinking maybe something like this

```ts
const [ pool, offset ] = getPool() // Get pool may create a new pool if necessary. It will not 
// perform checks and assign bytes to pool
...
incrementPool() // or something
```